extensions/jms/src/main/java/org/mule/extensions/jms/internal/connection/session/JmsSessionManager.java

```diff
@@ -126,21 +123,21 @@ public void recoverSession(String ackId) throws JMSException {
    * @param session session to bind
    */
   public void bindToTransaction(JmsSession session) {
-    currentSession.set(session);
+    getTransactionInformation().setJmsSession(session);
```

Comment:
you are assuming that jmsSession will be null in the TxInformation. Is that a fair assumption? What would happen if another session already there?